File: 1380. Lucky Numbers in a Matrix/solution.java
class Solution {
  public List<Integer> luckyNumbers (int[][] matrix) {
    int[] rows = new int[matrix.length];
    Arrays.fill(rows, 1000000);
    int[] columns = new int[matrix[0].length];
    Arrays.fill(columns, 0);
    for (int i = 0; i < matrix.length; i++){
      for (int j = 0; j < matrix[i].length; j++){
        if (matrix[i][j] < rows[i]){
          rows[i] = matrix[i][j];
        }
        if (matrix[i][j] > columns[j]){
          columns[j] = matrix[i][j];
        }
      }
    }

    for (int index = 0; index < rows.length; index++){
      int row = rows[index];
      for (int index2 = 0; index2 < columns.length; index2++){
        int column = columns[index2];
        if (row == column){
          return Arrays.asList(row);
        }
      }
    }
    return Arrays.asList();
  }
}

File: 1380. Lucky Numbers in a Matrix/solution.cpp
class Solution {
public:
  vector<int> luckyNumbers (vector<vector<int>>& matrix) {
    vector<int> rows(matrix.size(), 1000000);
    vector<int> columns(matrix[0].size(), 0);
    for (int i = 0; i < matrix.size(); i++){
      for (int j = 0; j < matrix[i].size(); j++){
        if (matrix[i][j] < rows[i]){
          rows[i] = matrix[i][j];
        }
        if (matrix[i][j] > columns[j]){
          columns[j] = matrix[i][j];
        }
      }
    }

    for (int index = 0; index < rows.size(); index++){
      int row = rows[index];
      for (int index2 = 0; index2 < columns.size(); index2++){
        int column = columns[index2];
        if (row == column){
          return {row};
        }
      }
    }
    return {};
  }
};